TELCODOCS-930: RN Document new features

ktothill opened the pull request:
Fixes: [TELCODOCS-930](https://issues.redhat.com/browse/TELCODOCS-930)

For: Version 4.12+

Doc Preview [here](https://56147--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes.html#ocp-4-12-0-ironic-agent-image-ztp)

Signed-off-by: Katie Tothill ktothill@redhat.com

ktothill commented:
CM email sent 2/20.

ktothill commented:
Awaiting merge of [PR 55899](https://github.com/openshift/openshift-docs/pull/55899) before rebasing and requesting merge to avoid conflicts.

oourfali commented:
looks good to me. Only question, regarding Nutanix, is whether we somehow need to mention it is SaaS only? We do mention UI + REST-API, but is that enough, or we should explicitly mention we're talking about the SaaS UI, rather than CIM?
@filanov will appreciate a review by you as well.

cybertron commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1269,6 +1269,27 @@ Optionally, you can now override the default evaluation intervals for all polici
+
+For more information about mirroring images, see xref:../installing/disconnected_install/installing-mirroring-installation-images.adoc#installation-mirror-repository_installing-mirroring-installation-images[Mirroring the OpenShift Container Platform image repository]
+
+[id="support-for-dual-stack-addressing-for-the-API-VIP-and-Ingress-VIP"]
```
Can we explicitly mention that in 4.12 dual stack is supported on baremetal only? We've gotten a few questions about that recently which seem to be related to a release note about the dual stack VIP feature. We added the ability to configure multiple VIPs for all platforms, but it's still only supported on baremetal for now.

ktothill replied:
Updated as suggested.

filanov commented:
+1 to @oourfali comment, currently it's supported only in SAAS, we plan to add the support for kube-api in 2.8

filanov commented:
There are few additional feature that we added and can be documented [MGMT-11058 Support setting kernel parameters for discovery](https://issues.redhat.com/browse/MGMT-11058)

ktothill commented:
Reached out to @johnwilkins and @rohennes to confirm answers to Oved, Ben and Michael's questions.

ktothill commented:
> There is additional feature that we added and can be documented [MGMT-11058 Support setting kernel parameters for discovery](https://issues.redhat.com/browse/MGMT-11058)

Added RN as requested.

ktothill commented:
Awaiting merge of https://github.com/openshift/openshift-docs/pull/55899 before rebasing and requesting merge to avoid conflicts.

ktothill commented:
/label merge-review-needed

ktothill commented:
@filanov @cybertron @oourfali @dtantsur can I please get an additional review and `LGTM` if applicable? Thank you!

racedo commented:
The Nutanix support should be removed from the list as this is only for the AI SaaS at the moment.

The heterogeneous cluster support is written as if it was only for the SaaS but it is for ACM too (MGMT-8890).

Since these release notes are for ACM they should include only ACM features.

ktothill commented:
@racedo, I've made the requested updates to the RN. Could you please review and provide feedback or an `LGTM` if applicable? Thank you!

cybertron commented:
/lgtm
for the note related to my area.

ktothill commented:
@racedo, I've made the requested updates to the RN. Could you please review and provide feedback or an LGTM if applicable? Thank you!

ktothill commented:
@chadcrum @trewest, can I please get a review and `LGTM` if applicable? Thank you!

adellape commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+
+* AArch64 Ironic agent image: `quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:cb0edf19fffc17f542a7efae76939b1e9757dc75782d4727fb0aa77ed5809b43`
+
+For more information about mirroring images, see xref:../installing/disconnected_install/installing-mirroring-installation-images.adoc#installation-mirror-repository_installing-mirroring-installation-images[Mirroring the OpenShift Container Platform image repository]
```
Punctuation

ktothill replied:
Updated as suggested.

adellape commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+
+The API VIP address and the Ingress VIP address must be of the primary IP address family when using dual-stack networking. Currently, Red Hat does not support dual-stack VIPs or dual-stack networking with IPv6 as the primary IP address family. However, Red Hat does support dual-stack networking with IPv4 as the primary IP address family. Therefore, you must place the IPv4 entries before the IPv6  ...
+
+[id="post-installation-configuration"]
```
Suggest a more unique / specific ID here.

danmacpherson commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+
+[id="post-installation-configuration"]
+= Assisted Installer SaaS provides platform integration support for Nutanix
+{ai-full} SaaS on link:https://console.redhat.com[console.redhat.com] supports installation of {product-title} on the Nutanix platform with Machine API integration using either the {ai-full} user interface or the REST API. Integration enables Nutanix Prism users to manage their infrastructure from a single interface, and enables auto-scaling. There are a few additional installation steps to enabl ...
```
Hi @racedo, I think if a release note is needed for MGMT-13374, it should go into the ACM release notes, not the OpenShift release notes. I'm also hesitant about putting AI SaaS release notes in the OpenShift docs since AI SaaS has it's own way of announcing features via the What's New.

ktothill commented:
/Label Telco
/Label Peer-Review-Needed

ktothill commented:
@chadcrum @trewest, can I please get a review and LGTM if applicable? Thank you!

racedo commented:
/lgtm thanks!

ktothill commented:
@chadcrum @trewest, can I please get a review and LGTM if applicable? Thank you!

ktothill commented:
/label remove peer-review-in-progress

chadcrum commented:
/lgtm

kcarmichael08 commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+For more information about mirroring images, see xref:../installing/disconnected_install/installing-mirroring-installation-images.adoc#installation-mirror-repository_installing-mirroring-installation-images[Mirroring the OpenShift Container Platform image repository].
+
+[id="support-for-dual-stack-addressing-for-the-API-VIP-and-Ingress-VIP"]
+= Support for dual-stack addressing for the API VIP and Ingress VIP
```
Checking that this is the right heading level - not sure if it's supposed to go under "new features and enhancements" in which case it would be `==`, I think? (I think the guideline is only one H1 per adoc file is allowed.)

ktothill replied:
Moved content and updated to H4.

kcarmichael08 reviewed (commented):
Couple of suggestions/questions; otherwise LGTM!

kcarmichael08 commented:
/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

ktothill commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+{ai-full} SaaS on link:https://console.redhat.com[console.redhat.com] supports installation of {product-title} on the Nutanix platform with Machine API integration using either the {ai-full} user interface or the REST API. Integration enables Nutanix Prism users to manage their infrastructure from a single interface, and enables auto-scaling. There are a few additional installation steps to enabl ...
+
+[id="configuring-kernel-arguments-for-the-Discovery-ISO-by-using-GitOps-ZTP"]
+= Configuring kernel arguments for the Discovery ISO by using GitOps ZTP
```
Moved content and updated to H4.

ktothill commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+
+[id="support-for-dual-stack-addressing-for-the-API-VIP-and-Ingress-VIP"]
+= Support for dual-stack addressing for the API VIP and Ingress VIP
+{ai-full} supports installation of {product-title} 4.12 and later versions with dual stack networking for the API VIP and Ingress VIP on bare metal only. This support introduces two new configuration settings: `api_vips` and `ingress_vips`, which can take a list of IP addresses. The legacy settings, `api_vip` and `ingress_vip` must also be set in {product-title} 4.12; however, since they only tak ...
```
Updated as suggested.

ktothill commented on release_notes/ocp-4-12-release-notes.adoc:
```diff
@@ -1289,6 +1289,40 @@ Any registry on this list is not required to have an entry in the pull secret us
+= Configuring kernel arguments for the Discovery ISO by using GitOps ZTP
+{product-title} now supports specifying kernel arguments for the Discovery ISO in GitOps ZTP deployments. In both manual and automated GitOps ZTP deployments, the Discovery ISO is part of the {product-title} installation process on managed bare-metal hosts. You can now edit the `InfraEnv` resource to specify kernel arguments for the Discovery ISO. This is useful for cluster installations with spe ...
+
+For more information about how to specify kernel arguments see xref:../scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-sites.adoc#setting-managed-bare-metal-host-kernel-arguments_ztp-deploying-far-edge-sites[Configuring kernel arguments for the Discovery ISO by using GitOps ZTP] and xref:../scalability_and_performance/ztp_far_edge/ztp-manual-install.adoc#setting-managed-bare-metal ...
```
Updated as suggested.

ktothill commented:
/label merge-review-needed
